Various tweaks

zoetrope69 opened the pull request:
+ Changes nav
+ Adds tabs instead of buttons
+ Moves search to by transactions list
+ Updates general style
+ Made it more mobile friendly
+ Put colours into variables
+ Move custom classmates into ClassNames pattern
+ Move the map to 'react-google-maps'
+ Map has improved pop-ups
+ Link between map page and list page and still have an item selected
+ Uses new Monzo API endpoints
+ Adds new categories

robcalcroft commented:
@zaccolley can you rebase pls, will review when I get a sec

robcalcroft commented:
Can you also squash all your commits into one with a message that follows the angular commit message style, thanks 😄

robcalcroft commented on src/vars.css:
```diff
@@ -1,20 +1,36 @@
+
+  --color-category-transport: #1E788C;
+  --color-category-groceries: #F5B946;
+  --color-category-eating_out: #E64B5F;
```
Can you camel case this?

zoetrope69 replied:
yes 🐫

robcalcroft commented on src/style.css:
```diff
@@ -1,40 +1,20 @@
-  font-family: 'Source Sans Pro', system-ui;
-  background-color: var(--color-off-white);
+  font-size: 18px;
+  font-family: 'Source Sans Pro', 'HelveticaNeue', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif;
```
Why did you remove `system-ui`?

zoetrope69 replied:
It varies a lot across operating systems, often doesn't consistent (e.g on ubuntu the system font is a lot different to source sans pro)  goodhttp://markdotto.com/2018/02/07/github-system-fonts/

robcalcroft replied:
Fair that makes sense 👍

robcalcroft commented on src/index.html:
```diff
@@ -6,6 +6,7 @@
   <meta http-equiv="X-UA-Compatible" content="ie=edge">
   <link rel="dns-prefetch shortcut icon" href="https://monzo.com/static/images/favicon.png">
   <title>Monzo Web</title>
+  <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:400,600" rel="stylesheet">
```
Wasn't this served locally before?

zoetrope69 replied:
Trying to get more control over what we need as here it's only the 400 and 600 weight

robcalcroft commented on src/components/TransactionSearch/style.css:
```diff
@@ -1,23 +1,18 @@
 
 .mzw-transaction-search__input {
-  padding: 0.5rem 1rem;
+  padding: 0.5em 1em;
```
Why are these `em` now?

zoetrope69 replied:
rem is relative to the root element so doesn't scale based on the other elements

robcalcroft commented on src/components/Map/style.css:
```diff
@@ -1,18 +1,41 @@
+  margin-top: 5px;
+  min-width: 250px;
+  font-size: 12px;
+  font-family: 'Source Sans Pro', 'HelveticaNeue', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif;
```
This should just be source sans and system ui as cssnext will transform it to the above

zoetrope69 replied:
system-ui turns into helvetica etc?

Commit:
refactor(everything): Various tweaks from styling to the map

Changes nav
Adds tabs instead of buttons
Moves search to by transactions list
Updates general style
Made it more mobile friendly
Put colours into variables
Move custom classmates into ClassNames pattern
Move the map to 'react-google-maps'
Map has improved pop-ups
Link between map page and list page and still have an item selected

zaccolley commented:
Thin I squashed it, not sure how I rebase. Do you want to fix anything from those other comments you left?

robcalcroft commented:
Squash looks good and nah I'm happy with everything 😄 will merge when you give the thumbs up

robcalcroft commented:
<img width="902" alt="screen shot 2018-03-06 at 23 18 33" src="https://user-images.githubusercontent.com/3504764/37064396-bda2c81e-2194-11e8-91d8-76eddb062b83.png">

Commit:
fix(layout): Map is now full height

zaccolley commented:
Fixed that 4 u rob

robcalcroft commented:
Sorry, another one: 
<img width="1050" alt="screen shot 2018-03-15 at 19 25 53" src="https://user-images.githubusercontent.com/3504764/37486648-07982ace-2887-11e8-8402-368186e71ba9.png">
Theres some extra space at the bottom of the transactions component and between the tabs at the top.

Commit:
Update Monzo API endpoints

Commit:
Fixed spacing of items

Commit:
Update and add in new categories

Commit:
Remove old font

zaccolley commented:
@robcalcroft Have fixed that and also added in the new categories

Commit:
Update screenshot
